### PR TITLE
[FIX] issue 999: error in generate.ts is masked and hence command lin…

### LINKED ISSUE #999: error in generate.ts is masked and hence command line is returning status 0
**Describe the bug:**
`generateConfig` function is returning err object and not throwing it. hence `execute` function does not have the chance to catch the exception and return 0.

**To Reproduce:**
Bug is found via code inspection and this problem can be simulated by unit test.
Unit test will be part of the PR to fix this problem to prevent regression.

**Expected behavior:**
command tool to return status code 1

**Screenshots:**

**Additional context:**

**Does this require updates to documentation?:**
No

### PR DESCRIPTION
**Pull request by dennisseah**
…e is returning status 0

Closes https://github.com/microsoft/bedrock/issues/999

modified code to not catch the exception and return it. Instead allow the exception to propagate to the calling function

### CODE REVIEW

**Review by andrebriggs — APPROVED** [review]

**Comment by andrebriggs** [comment]
@dennisseah Tests are failing

**Comment by dennisseah** [comment]
> @dennisseah Tests are failing

Yes, I need help from Nate.

**Comment by dennisseah** [comment]
> > @dennisseah Tests are failing
> 
> Yes, I need help from Nate.

we have figured out the problem and the problem is in the unit test and our implementation is masking the issue. Hence fixing the implementation causes problem because unit test is incorrect. I have moved the git clone tests to the beginning of the test run and hence `~/.spk/templates/_yradsmikham_spk_source/` is available in the tests later. tests in PR passed now.

File: src/commands/infra/generate.test.ts [hunk 1]
```diff
@@ -52,6 +52,147 @@ beforeEach(() => {
   jest.restoreAllMocks();
 });
 
+//////////////////////////////////////////////////////////////////////////////
```
Comment (NathanielRose) [inline src/commands/infra/generate.test.ts:55]: Did you mean to keep this?

**Review by NathanielRose — APPROVED** [review]
Integration tests and unit tests passing locally.

File: src/commands/infra/generate.test.ts (hunk 1 above)
Comment (dennisseah) [reply src/commands/infra/generate.test.ts:55]: to clearly separate git related tests from other tests

**Pull request merged**